src/components/SingleProductPage/SingleProduct.js

```diff
@@ -0,0 +1,79 @@
+import "./SingleProduct.css";
+import { useParams, useHistory } from "react-router";
+import useProducts from "../../hooks/useProducts";
+import getProductById from "../../utility/getProductById";
+import { FaRegSnowflake, FaArrowLeft } from "react-icons/fa";
+import { BsFillClockFill } from "react-icons/bs";
+
+export default function SingleProduct() {
+  const { singleProductId } = useParams();
+  const products = useProducts();
+  const history = useHistory();
+
+  if (products.length !== 0) {
+    const productInfo = getProductById(Number(singleProductId), products);
+    const {
+      name,
+      image,
+      ingredients,
+      time,
+      refrigeration,
+      goals,
+    } = productInfo;
+
+    const classForTonerImage =
+      name === "Glycolic Acid 7% Toning Solution" ? "toner-image" : "";
+    return (
```

Comment:
Hmmm, I'm not sure if this is the right way to get to the return. I mean, that the return is in the if statement.
I was like at the end of the file: "What, where does this `else { "loading" }` come from??"

Maybe you could do something like this:
```suggestion
  if (products.length === 0) {
  return <div>Loading</div>
  }
  
const productInfo = getProductById(Number(singleProductId), products);
const {
  name,
  image,
  ingredients,
  time,
  refrigeration,
  goals,
  } = productInfo;

const classForTonerImage =
  name === "Glycolic Acid 7% Toning Solution" ? "toner-image" : "";
return (
// return the jsx without the else at the end
```

I think this is clearer, and you can see easily what happen when some condition is true. It always returns `loading` when `products.length === 0` and the other stuff is not executed. If `products.length === 0` false, then the other after is executed.

Reply:
I see. I did this to avoid getting the error on first render, when products was still an empty array. Seems to be working fine like this though, and a bit cleaner 👌